wpilibj/src/main/java/edu/wpi/first/wpilibj/smartdashboard/SendableBuilderImpl.java

```diff
@@ -710,4 +744,21 @@ public void publishConstRaw(String key, String typestring, byte[] value) {
     property.m_pub.set(value);
     m_properties.add(property);
   }
+
+  @Override
+  public <T> void caching(
+      String key,
+      IntSupplier hashFunction,
+      Supplier<T> getter,
+      Consumer<T> setter,
+      PropertyFn<T> property) {
+    property.addProperty(key, getter, setter);
+
+    // Remove the property added above, wrap it, and add the wrapper in its place
+    var originalProperty = m_properties.get(m_properties.size() - 1);
+    var cachedProperty = new CachableProperty<>(originalProperty, hashFunction);
```

Comment:
This seems hacky and fragile, but all the `add*Property` methods will append a new property object to the list. However, if a user doesn't use one of those methods in the property function, then this would wrap an existing property. 

Should check that a new entry was appended by the call to `property.addProperty` before popping the last value off.